js/apps/admin-ui/src/realm-settings/OpenIdEndpointConigurationTab.tsx

```diff
@@ -0,0 +1,175 @@
+        role="manage-realm"
+        className="pf-u-mt-lg"
+        onSubmit={handleSubmit(save)}
+      >
+        <FormGroup
+          label={t("claimsSupported")}
+          fieldId="kc-claims-supported"
+          labelIcon={
+            <HelpItem
+              helpText={t("claimsSupportedHelp")}
+              fieldLabelId="claimsSupported"
+            />
+          }
+        >
+          <Controller
+            name="attributes"
+            control={control}
+            render={({ field }) => {
+              const claimsSupported = field.value?.claimsSupported?.trim()
+                ? field.value.claimsSupported.split(",")
+                : [];
+              return (
+                <>
+                  {claimsSupported.map((claim: string, index: number) => {
+                    if (claim) {
+                      return (
+                        <InputGroup key={index}>
+                          <TextInput
+                            id={"textInput" + index}
+                            value={claim}
+                            onChange={(value) => {
+                              claimsSupported[index] = value;
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            aria-label="input with button"
+                          />
+                          <Button
+                            id={"inputButtonRemove" + index}
+                            icon={<MinusIcon />}
+                            onClick={() => {
+                              claimsSupported.splice(index, 1);
+                              field.value &&
+                                (field.value.claimsSupported =
+                                  claimsSupported.join(","));
+                              field.onChange(field.value);
+                            }}
+                            variant="control"
+                          />
+                        </InputGroup>
+                      );
+                    }
+                  })}
+                  <InputGroup>
+                    <TextInput
+                      id="textInputAdd"
+                      value={newClaim}
+                      onChange={(value) => {
+                        setNewClaim(value);
+                      }}
+                      aria-label="input with button"
+                    />
+                    <Button
+                      id={"inputButtonAdd"}
+                      icon={<PlusIcon />}
+                      onClick={() => {
+                        if (newClaim) {
+                          claimsSupported.push(newClaim);
+                          field.value &&
+                            (field.value.claimsSupported =
+                              claimsSupported.join(","));
```

Comment:
This value should not be mutated in place, instead a new copy should be passed into `field.onChange()`.

Reply:
Now using 
```
let attributes = field.value || {};
claimsSupported.push(newClaim);
attributes.claimsSupported = claimsSupported.join(","); 
setNewClaim("");
field.onChange(attributes);
```